show/reboot_cause.py

```diff
@@ -62,33 +155,24 @@ def reboot_cause(ctx):
         click.echo(reboot_cause_str)
 
 
-# 'history' subcommand ("show reboot-cause history")
+# 'all' command within 'reboot-cause'
 @reboot_cause.command()
-def history():
-    """Show history of reboot-cause"""
-    REBOOT_CAUSE_TABLE_NAME = "REBOOT_CAUSE"
-    TABLE_NAME_SEPARATOR = '|'
-    db = SonicV2Connector(host='127.0.0.1')
-    db.connect(db.STATE_DB, False)   # Make one attempt only
-    prefix = REBOOT_CAUSE_TABLE_NAME + TABLE_NAME_SEPARATOR
-    _hash = '{}{}'.format(prefix, '*')
-    table_keys = db.keys(db.STATE_DB, _hash)
-    if table_keys is not None:
-        table_keys.sort(reverse=True)
+def all():
+    """Show cause of most recent reboot"""
+    reboot_cause_data = fetch_reboot_cause_from_db("all")
+    header = ['Device', 'Name', 'Cause', 'Time', 'User']
+    click.echo(tabulate(reboot_cause_data, header, numalign="left"))
 
-        table = []
-        for tk in table_keys:
-            entry = db.get_all(db.STATE_DB, tk)
-            r = []
-            r.append(tk.replace(prefix, ""))
-            r.append(entry['cause'] if 'cause' in entry else "")
-            r.append(entry['time'] if 'time' in entry else "")
-            r.append(entry['user'] if 'user' in entry else "")
-            r.append(entry['comment'] if 'comment' in entry else "")
-            table.append(r)
 
-        header = ['Name', 'Cause', 'Time', 'User', 'Comment']
-        click.echo(tabulate(table, header, numalign="left"))
+# 'history' command within 'reboot-cause'
+@reboot_cause.command()
+@click.argument('module_name', required=False)
+def history(module_name):
+    """Show history of reboot-cause"""
+    reboot_cause_history = fetch_reboot_cause_history_from_db(module_name)
+    if module_name is not None:
+        header = ['Device', 'Name', 'Cause', 'Time', 'User', 'Comment']
+        click.echo(tabulate(reboot_cause_history, header, numalign="left"))
     else:
-        click.echo("Reboot-cause history is not yet available in StateDB")
-        sys.exit(1)
+        header = ['Name', 'Cause', 'Time', 'User', 'Comment']
+        click.echo(tabulate(reboot_cause_history, header, numalign="left"))
```

Comment:
@rameshraghupathy please test this CLI on non-smartswitch

Reply:
Done and UT log is attached
@prgeor